business/apps.go

```diff
@@ -63,19 +210,63 @@ func (in *AppService) GetAppList(namespace string) (models.AppList, error) {
 			Name:         keyApp,
 			IstioSidecar: true,
 		}
-		labels := make(map[string][]string)
+		applabels := make(map[string][]string)
+		apVsNames := make([]string, 0)
+		apDrNames := make([]string, 0)
 		for _, srv := range valueApp.Services {
-			joinMap(labels, srv.Labels)
+			joinMap(applabels, srv.Labels)
+
+			svcVirtualServices := kubernetes.FilterVirtualServices(virtualServices, srv.Namespace, srv.Name)
+			apVsNames = append(apVsNames, getIstioResourcesNames(svcVirtualServices)...)
+
+			svcDestinationRules := kubernetes.FilterDestinationRules(destinationRules, srv.Namespace, srv.Name)
+			apDrNames = append(apDrNames, getIstioResourcesNames(svcDestinationRules)...)
 		}
+		appItem.VirtualServices = apVsNames
+		appItem.DestinationRules = apDrNames
+
+		apGwNames := make([]string, 0)
```

Comment:
Why don't have one single array of references (as done in the validations) {type,name,namespace?}? The frontend will be way easier to print. Removing switches and ifs.

In the validations there's this object:
https://github.com/kiali/kiali/blob/de7041d3c7b4b5e4b9185cea70572e092cd21318/models/istio_validation.go#L11-L15

And also, in the frontend there it is pretty simplified:
https://github.com/kiali/kiali-ui/blob/master/src/pages/IstioConfigDetails/ValidationReferences.tsx#L18-L28

Reply:
Oh, it makes sense, I missed that type, as it was named as "Validation" I missed for this.

Reply:
Yeah, feel free to move the struct into another module and maybe rename it. At the implementation time felt that it was the only moment when references were needed. So agree on adjusting this one.